c10/core/KinetoPluginAPI.cpp

```diff
@@ -0,0 +1,52 @@
+#ifdef USE_KINETO
+
+#include "KinetoPluginAPI.h"
+
+#include <map>
+#include <vector>
+#include <mutex>
+
+namespace c10 {
+namespace kineto_plugin {
+
+static std::map<DeviceType, std::vector<PluginFactory>> plugin_factories;
```

Comment:
use array

Reply:
> use array

array is not proper here, because we need to store the device associated register function here. We dont store register functions for all types.